### PR TITLE
DBZ-2938 Test stability

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.redhat.com/browse/DBZ-2938

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -63,25 +63,37 @@ public SnapshotResult execute(ChangeEventSourceContext context) throws Interrupt
...
             LOGGER.warn("Snapshot was interrupted before completion");
-            snapshotProgressListener.snapshotAborted();
             throw e;
         }
         catch (RuntimeException e) {
-            snapshotProgressListener.snapshotAborted();
+            completed = false;
             throw e;
         }
         catch (Throwable t) {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java:83]: While you are in the code, do we really wnat to catch Errors? Shouldn't we catch `Exception` only?
Maybe the `catch` could be simplified to a single branch with `instanceof` to handle interruped exception and the rest via encapuslation to `DebeziumException`. In that case the `completed` flag would be unnecessary

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java]: Yes, catching `Throwable` indeed doesn't sound right. Slightly reworked it, but the flag is still there. I don't think instanceof is really better.

**Comment by gunnarmorling** [comment]
@jpechane, ready for another look. Test failures are independent.

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks

**Pull request merged**